statsmodels/tsa/statespace/regime_switching/kim_filter.py

```diff
@@ -0,0 +1,635 @@
+import numpy as np
+from scipy.misc import logsumexp
+from scipy.optimize import linprog
```

Comment:
(I'm just looking at this comment.)
If there is a gain to using it, then it might be better to keep the code and make it optional and protect the import and usage. scipy 0.15 might not be very far in the future as a minimum.
In quantile regression we avoided going for a linprog solution, since it doesn't seem to provide much additional benefit and at that time linprog really was new.